src/identity-user-interfaces.ts

```diff
@@ -0,0 +1,75 @@
+import {
+    AllUserAttributes,
+    ConsentState,
+    IdentityApiData,
+    IdentityCallback,
+    Product,
+} from '@mparticle/web-sdk';
+import { SDKIdentityTypeEnum } from './identity.interfaces';
+import { MessageType } from './types.interfaces';
+import { BaseEvent } from './sdkRuntimeModels';
+
+// https://go.mparticle.com/work/SQDSDKS-5033
+// https://go.mparticle.com/work/SQDSDKS-6354
+export interface MParticleUser {
```

Comment:
The issue is that `User` and `MParticleUser` have some inconsistent method definitions and would require a larger initiative to resolve. SQDSDKS-5033 and SQDSDKS-6354 should address some of these issues, but I think we my have other sustained engineering tasks dedicated to fixing.

As a happy middle, I can extend rather than redefine the User, so we can be clear as to which methods are being overriden.

Reply:
I think this is a good idea, and we can keep the work item here as an optimization point in the future.